codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerCodegenVisitor.kt

```diff
@@ -444,3 +463,48 @@ open class ServerCodegenVisitor(
             .render()
     }
 }
+
+fun renderDefaultBuilder(model: Model, symbolProvider: RustSymbolProvider, member: MemberShape): Writable {
+    return writable {
+        val node = member.expectTrait<DefaultTrait>().toNode()!!
+        when (val target = model.expectShape(member.target)) {
+            is EnumShape, is IntEnumShape -> {
+                val value = when (target) {
+                    is IntEnumShape -> node.expectNumberNode().value
+                    else -> node.expectStringNode().value
+                }
+                val enumValues = when (target) {
+                    is IntEnumShape -> target.enumValues
+                    is EnumShape -> target.enumValues
+                    else -> mapOf<String, String>()
+                }
+                val variant = enumValues
+                    .entries
+                    .filter { entry -> entry.value == value }
+                    .map { entry ->
+                        symbolProvider.toEnumVariantName(
+                            EnumDefinition.builder().name(entry.key).value(entry.value.toString()).build(),
+                        )!!
+                    }
+                    .first()
+                val symbol = symbolProvider.toSymbol(target)
+                rust(".unwrap_or($symbol::${variant.name})")
+            }
+            is IntegerShape -> rust(".unwrap_or(${node.expectNumberNode().value})")
+            is FloatShape -> rust(".unwrap_or(${node.expectNumberNode().value.toFloat()})")
+            is BooleanShape -> rust(".unwrap_or(${node.expectBooleanNode().value})")
+            is StringShape -> rust(".unwrap_or_else(|| String::from(${node.expectStringNode().value.dq()}))")
+            is ListShape, is MapShape -> rust(".unwrap_or_default()")
+            is DocumentShape -> {
+                when (node) {
+                    is NullNode -> rust(".unwrap_or_default()")
+                    is BooleanNode -> rust(".unwrap_or(${node.value})")
+                    is StringNode -> rust(".unwrap_or_else(|| String::from(${node.value.dq()}))")
+                    is NumberNode -> rust(".unwrap_or(${node.value})")
+                    else -> rust(".unwrap_or_default()")
+                }
+            }
```

Comment:
This can be removed entirely. The `default` trait cannot be applied to Smithy document shapes.

Reply:
(as above) [here](https://smithy.io/2.0/spec/type-refinement-traits.html#default-trait) I read:

> document: can be set to null, `true, false, string, numbers, an empty list, or an empty map.

Reply:
You're right! My bad, I only read the trait selector; I didn't now `document` shapes were simple types.